src/events.rs

```diff
@@ -2,6 +2,17 @@ use std::path::PathBuf;
 
 #[derive(Clone, Debug)]
 pub enum Event {
+    WindowEvent {
```

Comment:
This variant could probably just be `Window` as we already know that it is an `Event` via the enum name, e.g. `Event::Window`. Otherwise we end up repeating "Event" twice.